galasa-extensions-parent/dev.galasa.auth.couchdb/src/main/java/dev/galasa/auth/couchdb/internal/CouchdbAuthStoreRegistration.java

```diff
@@ -0,0 +1,94 @@
+import dev.galasa.extensions.common.couchdb.CouchdbValidator;
+import dev.galasa.extensions.common.impl.HttpClientFactoryImpl;
+import dev.galasa.extensions.common.impl.HttpRequestFactoryImpl;
+import dev.galasa.extensions.common.api.HttpRequestFactory;
+import dev.galasa.extensions.common.impl.LogFactoryImpl;
+import dev.galasa.framework.spi.IFrameworkInitialisation;
+import dev.galasa.framework.spi.SystemEnvironment;
+import dev.galasa.framework.spi.auth.IAuthStoreRegistration;
+import dev.galasa.framework.spi.utils.SystemTimeService;
+import dev.galasa.framework.spi.auth.AuthStoreException;
+
+@Component(service = { IAuthStoreRegistration.class })
+public class CouchdbAuthStoreRegistration implements IAuthStoreRegistration {
+
+    private HttpClientFactory httpClientFactory;
+    private HttpRequestFactory httpRequestFactory;
+    private LogFactory logFactory;
+    private CouchdbValidator couchdbValidator;
+
+    public CouchdbAuthStoreRegistration() {
+        this(
+            new HttpClientFactoryImpl(),
+            new HttpRequestFactoryImpl(CouchdbAuthStore.COUCHDB_AUTH_TYPE, new SystemEnvironment().getenv(CouchdbAuthStore.COUCHDB_AUTH_ENV_VAR)),
+            new LogFactoryImpl(),
+            new CouchdbAuthStoreValidator()
+        );
+    }
+
+    public CouchdbAuthStoreRegistration(HttpClientFactory httpClientFactory, HttpRequestFactory httpRequestFactory, LogFactory logFactory, CouchdbValidator couchdbValidator) {
+        this.httpClientFactory = httpClientFactory;
+        this.httpRequestFactory = httpRequestFactory;
+        this.logFactory = logFactory;
+        this.couchdbValidator = couchdbValidator;
+    }
+
+    /**
+     * This method checks that the auth store is a remote URL reference, and if true
+     * registers a new couchdb auth store as the only auth store.
+     *
+     * @param frameworkInitialisation Parameters this extension can use to to
+     *                                initialise itself.
+     * @throws AuthStoreException if there was a problem accessing the auth store.
+     */
+    @Override
+    public void initialise(IFrameworkInitialisation frameworkInitialisation) throws AuthStoreException {
+
+        URI authStoreUri = frameworkInitialisation.getAuthStoreUri();
+
+        if (isUriReferringToThisExtension(authStoreUri)) {
+            try {
+                frameworkInitialisation.registerAuthStore(
+                    new CouchdbAuthStore(
+                        authStoreUri,
+                        httpClientFactory,
+                        httpRequestFactory,
+                        logFactory,
+                        couchdbValidator,
+                        new SystemTimeService()
+                    )
+                );
+            } catch (CouchdbException e) {
+                String errorMessage = ERROR_FAILED_TO_INITIALISE_AUTH_STORE.getMessage(e.getMessage());
+                throw new AuthStoreException(errorMessage, e);
```

Comment:
We could log the message before we throw, in case it gets lost ?

Reply:
Added error message logging to the AuthStoreException constructor